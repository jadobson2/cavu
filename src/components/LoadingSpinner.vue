<template>
  <div
    class="w-14 h-14 my-14 mx-auto"
    :class="{ 'invert': invert }"
  >
    <img
      src="../assets/icons/spinner.svg"
      alt="Loading spinner"
      class="animate-spin"
    />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'LoadingSpinner',
  props: {
    invert: {
      type: Boolean,
      default: false
    }
  }
})
</script>
