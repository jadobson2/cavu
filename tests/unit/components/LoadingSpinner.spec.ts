import { shallowMount } from '@vue/test-utils'
import LoadingSpinner from '@/components/LoadingSpinner.vue'

describe('LoadingSpinner', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(LoadingSpinner)

    expect(wrapper.element).toMatchSnapshot()
  })

  test('invert', () => {
    const wrapper = shallowMount(LoadingSpinner, {
      propsData: { invert: true }
    })

    expect(wrapper.classes()).toContain('invert')
  })
})
