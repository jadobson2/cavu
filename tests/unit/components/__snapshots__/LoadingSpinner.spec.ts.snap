// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly 1`] = `
<div
  class="w-14 h-14 my-14 mx-auto"
>
  <img
    alt="Loading spinner"
    class="animate-spin"
    src="../assets/icons/spinner.svg"
  />
</div>
`;
